plugin/environment.go

```diff
@@ -28,7 +28,7 @@ type apiImplCreatorFunc func(*model.Manifest) API
 // If an installed plugin is missing from the env.registeredPlugins map, then the
 // plugin is configured as disabled and has not been activated during this server run.
 type registeredPlugin struct {
-	BundleInfo *model.BundleInfo
+	BundleInfo *model.BundleInfo // This should not be modified once accessed outside the registeredPlugins map.
```

Comment:
I wanted to express the point that since this is a pointer, irrespective of the fact whether it is exported or not, once it is returned from a method to an external package, it should not be modified. I am fine with unexporting it. But where do you suggest I should place the comment, or you feel this is not required ?

Reply:
Would the same comment be necessary on `failTimeStamps` and `supervisor` since they are pointers? If there's some distinction that makes `BundleInfo` special, then I can better appreciate the comment. but if not -- and especially if unexported -- it does feel like the comment might be unnecessary.

Reply:
None of them are actually being returned outside the package. But the `Active` method actually returns elements of the slice which are active. Hence my concern.

Reply:
Understood, that's a good point. It doesn't seem likely that a client of `Active` would ever encounter this particular comment. Should we move the immutability concern to the documentation of `Active` instead?

Reply:
Fair enough, will change.